app/controllers/provider/admin/cms/visit_portal_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class Provider::Admin::CMS::VisitPortalController < Provider::Admin::CMS::BaseController
+  # Encrypt the CMS token under a temporary SSO token and redirect to the Developer Portal
+  def with_token
+    verifier = Rails.application.message_verifier(:cms_token)
+    cms_token = current_account.settings.cms_token!
+    expires_at = Time.now.utc.round + 1.minute
+    signing_options = { purpose: :cms_edit_mode, expires_at: }
+    signature = verifier.generate(cms_token, **signing_options).split("--").last
```

Comment:
I think we can just ignore the actual `cms_token`. As a message the account id can be used. Or anything else identifying the portal. Then on verification just verify user is trying to edit the correct portal (as in the signed message) and is within the expiration time.

Reply:
Yeah we could do that but currently I don't see the value on the change. The current code is completely based on the CMS token, for instance see how `handle_cms_token` behaves based on the token:

https://github.com/3scale/porta/blob/3b6dfb33332cf03015d3a0c6e4c9faaaacf4d893/lib/developer_portal/lib/cms/toolbar.rb#L19-L34

So:
- Valid token -> enable edit mode
- Invalid token -> disable edit mode
- Empty token -> Do nothing

And then the token is stored in the session, etc. All this logic should be removed, tests redone, etc. And the token is already perfectly valid to identify the portal so I don't see the need for the rework other than simplifying the code.

If you think it's a nice to do, please create a Jira issue and I'll do it eventually in another PR.

Reply:
> And then the token is stored in the session

:face_with_open_eyes_and_hand_over_mouth:

Reply:
We store the session in cookies but it's encrypted AFAIK

Reply:
Ok, see if https://issues.redhat.com/browse/THREESCALE-11466 sounds good to you.